azure/examples/bucket/index.ts

```diff
@@ -0,0 +1,21 @@
+// Copyright 2016-2018, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import * as cloud from "@pulumi/cloud";
+import * as pulumi from "@pulumi/pulumi";
+
+const bucket = new cloud.Bucket("myBucket");
+bucket.onPut("myPutHandler", async (args) => {
+    console.log(JSON.stringify(args, null, 2));
```

Comment:
Can we just use `context.log` which is supported?

Reply:
Not exactly.  Remember, this is in the onPut *callback*.  onPut simply passes data about the object that was put to the callback, there is no way to get to that context object here.

Now, we have some options:

1. we could pass this context object along as another arg to the callback.  This would be leaky, and would mean your code was now doing very azure specific stuff.  I don't like this.
2. we could attempt to intercept console events and then forward them to context.log ourself.  I think this would work.  What do you think?

Reply:
Ok.  I've gone with intercepting logging and just redirecting it to context logging.